modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -5088,13 +5108,23 @@ private void unlockListenerReadLock() {
 
     /** {@inheritDoc} */
     @Override public String toString() {
-        lockEntry();
-
-        try {
-            return S.toString(GridCacheMapEntry.class, this);
+        if (tryLockEntry(ENTRY_LOCK_TIMEOUT)) {
+            try {
+                return S.toString(GridCacheMapEntry.class, this);
+            }
+            finally {
+                unlockEntry();
+            }
         }
-        finally {
-            unlockEntry();
+        else {
+            return "GridCacheMapEntry [err='Partial result represented because entry lock wasn't acquired."
+                +" Waiting time elapsed.'"
+                + ", key=" + key
```

Comment:
Only key and hash are final here, access to other fields isn't safe.
Also, we pring key regardless of GridToStringBuilder#includeSensitive value.
Let's print only key and add check for sensitive mode.

Reply:
We can do this in superclass

    public String toStringWithTryLock(Supplier<String> defaultToString) {
        if (tryLockEntry(ENTRY_LOCK_TIMEOUT)) {
            try {
                return defaultToString.get();
            }
            finally {
                unlockEntry();
            }
        }
        else {
            return "Short representation";
        }
    }

and this in subclasses

    /** {@inheritDoc} */
    @Override public String toString() {
        return toStringWithTryLock(() -> S.toString(GridDhtCacheEntry.class, this,
            "part", locPart.id(),
            "super", super.toString()));
    }